main/modes/games/2048/mode_2048.c

```diff
@@ -0,0 +1,492 @@
+            while (checkButtonQueueWrapper(&evt))
+            {
+                if (evt.down)
+                {
+                    soundPlaySfx(&t48->click, MIDI_SFX);
+                    t48_gameInit(t48);
+                    t48->state = T48_IN_GAME;
+                }
+            }
+            // Draw
+            t48_drawStartScreen(t48, t48_generateRainbow());
+            t48_chaseLEDs(elapsedUs);
+            break;
+        }
+        case T48_WIN_SCREEN:
+        {
+            // Get inputs
+            while (checkButtonQueueWrapper(&evt))
+            {
+                if (evt.down && (evt.button & PB_A || evt.button & PB_B))
+                {
+                    t48->state = T48_IN_GAME;
+                }
+            }
+            fillDisplayArea(0, 0, TFT_WIDTH, TFT_HEIGHT, c000);
+            drawText(&t48->titleFont, c055, youWin, (TFT_WIDTH - textWidth(&t48->titleFont, youWin)) / 2, 48);
+            drawText(&t48->font, c555, continueAB, (TFT_WIDTH - textWidth(&t48->font, continueAB)) / 2,
+                     TFT_HEIGHT - 64);
+            break;
+        }
+        case T48_HS_SCREEN:
+        {
+            // Handle input in text entry object
+            while (checkButtonQueueWrapper(&evt))
+            {
+                t48->textEntryDone = !textEntryInput(evt.down, evt.button);
+            }
+
+            // If the text entry is done, sort the scores, reset the text entry object,
+            // and display the final screen
+            if (t48->textEntryDone)
+            {
+                t48SortHighScores();
+                textEntrySoftReset();
+                t48->state = T48_END_SCREEN;
+            }
+
+            // Draw text entry
+            textEntryDraw(elapsedUs);
+            break;
+        }
+        case T48_END_SCREEN:
+        {
+            // Check any button is pressed
+            while (checkButtonQueueWrapper(&evt))
+            {
+                if (evt.down && (evt.button & PB_A || evt.button & PB_B))
+                {
+                    soundPlaySfx(&t48->click, MIDI_SFX);
+                    t48->state = T48_START_SCREEN;
+                }
+            }
+
+            // Draw the final score screen
+            t48_drawGameOverScreen(t48, t48->score, t48_generateRainbow());
+            t48_chaseLEDs(elapsedUs);
+            break;
+        }
+        default:
+        {
+            break;
+        }
+    }
+}
+
+/**
+ * @brief Restarts teh BGM once the track ends
+ *
+ */
+static void t48BgmCb()
+{
+    t48->bgmIsPlaying = false;
+}
+
+/**
+ * @brief Initializes the high scores based either from NVS or predetermined scores to beat
+ *
+ */
+static void t48InitHighScores()
+{
+    // Init High scores
+    for (int8_t i = 0; i < T48_HS_COUNT; i++)
+    {
+        if (!readNvs32(highScoreKey[i], &t48->highScore[i]))
+        {
+            switch (i)
+            {
+                case 0:
+                    t48->highScore[i] = 96880;
```

Comment:
Randomly chosen? Maybe make them all unlikely-even numbers so it's easier for players to tell which are preloaded and which are their scores? like 96000 / 69000 / 24000 / 11000 / 5000?

Reply:
Actual real scores I achieved while testing. I wanted them to feel like they weren't faked.

Reply:
Fair enough!